lib/move/move.js

```diff
@@ -18,24 +18,25 @@ function move (src, dest, opts, cb) {
 
   stat.checkPaths(src, dest, 'move', opts, (err, stats) => {
     if (err) return cb(err)
-    const { srcStat } = stats
+    const { srcStat, isChangingCase = false } = stats
     stat.checkParentPaths(src, srcStat, dest, 'move', err => {
       if (err) return cb(err)
       mkdirp(path.dirname(dest), err => {
         if (err) return cb(err)
-        return doRename(src, dest, overwrite, cb)
+        return doRename(src, dest, overwrite, isChangingCase, cb)
       })
     })
   })
 }
 
-function doRename (src, dest, overwrite, cb) {
+function doRename (src, dest, overwrite, isChangingCase, cb) {
   if (overwrite) {
     return remove(dest, err => {
```

Comment:
If `overwrite: true` and the file just changes case, won't it then remove the file that it's trying to rename?

Reply:
Yeah I see your point! It is safer to check changing case before checking overwrite. However, I don't understand what would be the point of using `overwrite: true` when you just want to change the case!

Basically, the `overwrite: true` is useful when there is a possibility (or we know for sure) that dest already exists and we want to move src to dest regardless. In this case, we are dealing with 2 files and `overwrite: true` is needed in order to get the desired result. But, when we are just changing the case, we are basically dealing with 1 file, so there is no need to use the `overwrite` option!

Reply:
I can see this happening when a dev just wants to move a file to where some result says it should, and the result just changes the case.

I saw that you moved out, but maybe this is important enough to warrant a test?

Reply:
I verified that on Mac, but it doesn't hurt to have a test for it!